rskj-core/src/main/java/org/ethereum/rpc/dto/BlockResultDTO.java

```diff
@@ -131,10 +132,22 @@ public static BlockResultDTO fromBlock(Block b, boolean fullTx, BlockStore block
         List<Transaction> blockTransactions = b.getTransactionsList();
         if (fullTx) {
             for (int i = 0; i < blockTransactions.size(); i++) {
+                Transaction tx = blockTransactions.get(i);
+
+                if (skipRemasc && tx.isRemascTransaction(i, blockTransactions.size())) {
```

Comment:
mmm we don't want to skip remasc, I'd rather be more generic, we want to skip tx with gas used = 0. I know remasc is the only one that has gas used = 0, but this should be refactored. You can filter before going into the `for` instead of mantaining duplicated logic. And even if we just want to skip remasc it should be done outside `for` cycle

Reply:
maybe
```java
List<Transaction> blockTransactions = withGasPriceGreaterThanZero(b.getTransactionsList(), flag or skipRemasc)
```